client/my-sites/site-settings/form-general.jsx

```diff
@@ -286,71 +357,73 @@ const FormGeneral = React.createClass( {
 						<FormRadio
 							name="blog_public"
 							value="-1"
-							checked={ -1 === parseInt( this.state.blog_public, 10 ) }
+							checked={ -1 === parseInt( fields.blog_public, 10 ) }
 							onChange={ this.handleRadio }
-							disabled={ this.state.fetchingSettings }
+							disabled={ isRequestingSettings }
 							onClick={ this.onRecordEvent( 'Clicked Site Visibility Radio Button' ) } />
-						<span>{ this.translate( 'Private' ) }</span>
+						<span>{ translate( 'Private' ) }</span>
 						<FormSettingExplanation isIndented>
-							{ this.translate( 'Your site is only visible to you and users you approve.' ) }
+							{ translate( 'Your site is only visible to you and users you approve.' ) }
 						</FormSettingExplanation>
 					</FormLabel>
 				}
 
 			</FormFieldset>
 		);
-	},
+	}
 
-	handleAmpToggle() {
-		this.setState( { amp_is_enabled: ! this.state.amp_is_enabled }, () => {
+	handleAmpToggle = () => {
+		const { fields, updateFields } = this.props;
+		updateFields( { amp_is_enabled: ! fields.amp_is_enabled }, () => {
 			this.submitForm();
-			this.onRecordEvent( 'Clicked AMP Toggle' );
+			this.onRecordEvent( 'Clicked AMP Toggle' )();
```

Comment:
Yes, I agree this syntax is weird. I think building a partial function in mapDispatchToProps is a good idea to avoid it, but we'll still need a `onRecordEvent` function to factorize using this as a callback.

```js
<FormInput
	onClick={ this.onRecordEvent( 'Clicked Site Title Field' ) } 
/>
```